app/org/sagebionetworks/bridge/models/schedules/ActivityScheduler.java

```diff
@@ -64,7 +64,9 @@ protected void addScheduledActivityAtTime(List<ScheduledActivity> scheduledActiv
                 for (Activity activity : schedule.getActivities()) {
                     ScheduledActivity schActivity = ScheduledActivity.create();
                     schActivity.setSchedulePlanGuid(plan.getGuid());
-                    schActivity.setTimeZone(context.getZone());
+                    //schActivity.setTimeZone(context.getZone());
+                    // Use the time zone of the request, not the time zone we've been sequencing against. 
+                    schActivity.setTimeZone(context.getEndsOn().getZone());
```

Comment:
Also, a few test cases

1. Scheduler is interpreting the event to the correct date. Examples:
* Enrollment=1487552400000 (2017-02-20T01:00Z), timeZone=-08:00, endsOnTimeZone=-08:00, schedule=daily at 11pm, expected=2017-02-19T23:00-0800
* Enrollment=1487552400000 (2017-02-20T01:00Z), timeZone=+09:00, endsOnTimeZone=+09:00, schedule=daily at 11pm, expected=2017-02-20T23:00+0900

2. Scheduler is interpreting the event to the right local time. Examples:
* Enrollment=1487552400000 (2017-02-20T01:00Z), timeZone=-08:00, endsOnTimeZone=-08:00, schedule=one hour after enrollment, expected=2017-02-19T18:00-0800
* Enrollment=1487552400000 (2017-02-20T01:00Z), timeZone=+09:00, endsOnTimeZone=+09:00, schedule=one hour after enrollment, expected=2017-02-20T11:00+0900

3. Local times having timezone correctly applied. Examples:
* Enrollment=1487552400000 (2017-02-20T01:00Z), timeZone=-08:00, endsOnTimeZone=+09:00, schedule=one hour after enrollment, expected=2017-02-19T18:00+0900 (Should this be the 19th or the 20th?)
* Enrollment=1487552400000 (2017-02-20T01:00Z), timeZone=+09:00, endsOnTimeZone=-08:00, schedule=one hour after enrollment, expected=2017-02-20T11:00-0800

Are these test cases represented in the unit tests?